File: src/main/java/com/dunamismax/site/util/DateFormatting.java
package com.dunamismax.site.util;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Locale;
import org.springframework.stereotype.Component;

@Component("dates")
public class DateFormatting {

  private final DateTimeFormatter shortFormat = DateTimeFormatter.ofPattern("MMM d, yyyy", Locale.ENGLISH);
  private final DateTimeFormatter longFormat = DateTimeFormatter.ofPattern("MMMM d, yyyy", Locale.ENGLISH);

  public String shortDate(LocalDate date) {
    return date.format(shortFormat);
  }

  public String longDate(LocalDate date) {
    return date.format(longFormat);
  }
}

File: src/main/kotlin/com/dunamismax/site/util/DateFormatting.kt
package com.dunamismax.site.util

import org.springframework.stereotype.Component
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.Locale

@Component("dates")
class DateFormatting {
    private val short = DateTimeFormatter.ofPattern("MMM d, yyyy", Locale.ENGLISH)
    private val long = DateTimeFormatter.ofPattern("MMMM d, yyyy", Locale.ENGLISH)

    fun short(date: LocalDate): String = date.format(short)
    fun long(date: LocalDate): String = date.format(long)
}
